src/hooks/things.ts

```diff
@@ -156,7 +157,7 @@ export function useThingInResource(
   const thing = useMemoCompare(thisThing, dequal);
   const saveThing = useCallback(
     async (newThing: Thing) => {
-      const newDataset = setThing(resource || createSolidDataset(), newThing);
```

Comment:
I think we still need a mechanism to let people create a thing that doesn't exist yet.

since we can't distinguish between "the resource doesn't exist" and "the resource hasn't loaded yet" I think we need to allow callers to pass in a resource (which could be newly created) to use.

this still needs testing, but the idea is implemented here: https://github.com/mysilio-co/swrlit/pull/3

Reply:
They can call `saveResource` with the empty dataset. We could do a 404 check here like we do elsewhere, but would still need to error on nil-but-no-404.

Reply:
yea, i think the 404 check is probably right - agree re: nil-but-no-404